components/BottomSheetActions.tsx

```diff
@@ -0,0 +1,116 @@
+import React from "react";
+import { Modal, Pressable, StyleSheet, View, Text } from "react-native";
+import { List } from "react-native-paper";
+import { useRouter } from "expo-router";
+import BottomSheet, { BottomSheetView } from "@gorhom/bottom-sheet";
+import { useAuthContext } from "@/contexts";
+
+interface BottomSheetActionsProps {
+  cardType: "influencerType" | "promotionType";
+  cardId?: string;
+  isVisible: boolean;
+  snapPointsRange: [string, string];
+  onClose: () => void;
+}
+
+const BottomSheetActions = ({
+  cardType,
+  cardId,
+  isVisible,
+  snapPointsRange,
+  onClose,
+}: BottomSheetActionsProps) => {
+  const router = useRouter();
+  const sheetRef = React.useRef<BottomSheet>(null);
+
+  // Adjust snap points for the bottom sheet height
+  const snapPoints = React.useMemo(
+    () => [snapPointsRange[0], snapPointsRange[1]],
+    []
+  );
+
+  const handleClose = () => {
+    if (sheetRef.current) {
+      sheetRef.current.close();
+    }
+    onClose(); // Close the modal after the bottom sheet closes
+  };
+
+  const renderContent = () => {
```

Comment:
This can be a component (e.g. BottomSheetContent) instead of a function. And we can keep this inside a folder (components/BottomSheet - BottomSheetContent, BottomSheetActions)